BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void Or(BitList bits)
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            this.bits.Or(bits.bits);
+    }
+
+    /// <summary>
+    /// XOR boolean operation in this instance.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public bool Xor()
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+        {
+            bool result = bits[0];
+            for (int i = 1; i < bits.Length; i++)
+                result ^= bits[i];
+            return result;
+        }
+    }
+
+    /// <summary>
+    /// XOR boolean operation with another list of bit.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void Xor(BitList bits)
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            this.bits.Xor(bits.bits);
+    }
+
+    /// <summary>
+    /// Add a list of bit to the last position of this instance.
+    /// </summary>
+    /// <param name="items">A list of bit.</param>
+    public void Add(BitList items)
+    {
+        var newBitList = IsEmpty ? items : this + items;
+        this.bits = newBitList.bits;
+    }
+
+    /// <summary>
+    /// Add a bit to the last position of this instance.
+    /// </summary>
+    /// <param name="item">A bit.</param>
+    public void Add(bool item)
+    {
+        var newBitList = IsEmpty ? new BitList(item) : this + item;
+        bits = newBitList.bits;
+    }
+
+    /// <summary>
+    /// Add a list of bit to the first position of this instance.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public void AddFirst(BitList items)
+    {
+        Insert(0, items);
+    }
+
+    /// <summary>
+    /// Add a bit to the first position of this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public void AddFirst(bool item)
+    {
+        Insert(0, item);
+    }
+
+    /// <summary>
+    /// Remove a bit in the specified position.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void RemoveAt(int index)
+    {
+        if (!IsEmpty && index < Count && index >= 0)
+        {
+            var bools = ToBooleanArray().ToList();
+            bools.RemoveAt(index);
+            bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new ArgumentOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Remove a range of bits from this instance.
+    /// </summary>
+    /// <param name="index">Start index.</param>
+    /// <param name="count">Count of bits.</param>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void RemoveRange(int index, int count)
+    {
+        if (!IsEmpty && index < bits.Length && index >= 0 && count <= bits.Length - index && count > 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)